File: modules/01-basic-syntax/typescript/src/modules/compound_types.ts
import * as utils from '../utils/utils';

// Array assignment and access
export function array_asignment_and_access() {
  utils.print_h3('Array assignment and access');

  let my_ints_array: Array<number> = [];
  for (let i=0; i<10; i++) {
    my_ints_array.push(42);
  }
  my_ints_array[5] = 0;
  console.log(`my_ints_array: ${my_ints_array}`);
}

export function tuple_assignment_and_access() {
  utils.print_h3('Tuple assignment and access');

  const my_mixed_types_tuple = [7, true];
  console.log(`1st index: ${my_mixed_types_tuple[0]}`);
  console.log(`2nd index: ${my_mixed_types_tuple[1]}`);
}

File: modules/01-basic-syntax/rust/src/modules/compound_types.rs
use crate::utils;

// Array assignment and access
pub fn array_asignment_and_access() {
    utils::print_h3("Array assignment and access");

    let mut my_ints_array: [i8; 10] = [42; 10];
    my_ints_array[5] = 0;
    println!("my_ints_array: {my_ints_array:?}");
}

// Tuple assignment and access
pub fn tuple_assignment_and_access() {
    utils::print_h3("Tuple assignment and access");

    let my_mixed_types_tuple: (i8, bool) = (7, true);
    println!("1st index: {}", my_mixed_types_tuple.0);
    println!("2nd index: {}", my_mixed_types_tuple.1);
}
